### PR TITLE
Added Extra helper functions

### PR DESCRIPTION
**Pull request by dganzella**
Functions that were missing from the Animation system, it makes it more complete now

### CODE REVIEW

**Comment by Whitebrim** [comment]
Who is debugchicken?

**Comment by dganzella** [comment]
> Who is debugchicken?

its me!

unless thats a rhetorical question

**Comment by Whitebrim** [comment]
I'll take a closer look this weekend

**Comment by Whitebrim** [comment]
`.states[]` is a public property, you can access it from outside.
https://github.com/Whitebrim/AnimatedSprite/wiki/Examples#example_3b

File: AnimatedSprite.lua [hunk 1]
```diff
@@ -472,4 +472,30 @@ end
 
 function AnimatedSprite:update()
 	self:updateAnimation()
-end
\ No newline at end of file
+end
+
+--Extra helper functions
+
```
Comment (Whitebrim) [inline AnimatedSprite.lua]:
```suggestion
end

```

File: AnimatedSprite.lua [hunk 2]
```diff
@@ -472,4 +472,30 @@ end
...
+--Gets a state by name, so data like tick step can be changed dinamically, for example, in the situation of changing the speed of a spinning top
+---@return table
+function AnimatedSprite:getState(name)
+	return self.states[name]
+end
+
```
Comment (Whitebrim) [inline AnimatedSprite.lua]: Use `self.states.name` in your class. [Example](https://github.com/Whitebrim/AnimatedSprite/wiki/Examples#example_3b)

```suggestion
```

File: AnimatedSprite.lua [hunk 3]
```diff
@@ -472,4 +472,30 @@ end
...
+--gets the current state
+---@return table
```
Comment (Whitebrim) [inline AnimatedSprite.lua]: Comments use EmmyLua style, so three `-` is required
```suggestion
---Get the current state
---@return table state Reference to the current state
```

File: AnimatedSprite.lua [hunk 4]
```diff
@@ -472,4 +472,30 @@ end
...
+function AnimatedSprite:getCurrentState()
+	return self.states[self.currentState]
+end
+
+---gets the name of the current state, saves you having to keep track of it manually
+---@return string
+function AnimatedSprite:getCurrentStateName()
+	return self.currentState
+end
```
Comment (Whitebrim) [inline AnimatedSprite.lua]: Just use `self.currentState` in your class. [Example](https://github.com/Whitebrim/AnimatedSprite/wiki/Examples#example_3a)

```suggestion
```

File: AnimatedSprite.lua [hunk 5]
```diff
@@ -472,4 +472,30 @@ end
...
+
+---Get the current index in the array animation (aka, 1,2,3,n, where n = number of frames). good for starting another animation of same size from the same position of this animation
```
Comment (Whitebrim) [inline AnimatedSprite.lua]: Added more clarity
```suggestion
---Get the current frame's local index in the state (i.e. 1, 2, 3, n. where n = number of frames in this state)
```

File: AnimatedSprite.lua [hunk 6]
```diff
@@ -472,4 +472,30 @@ end
...
+---@return integer
+function AnimatedSprite:getCurrentFrameIndexInAnimationArray()
```
Comment (Whitebrim) [inline AnimatedSprite.lua]:
```suggestion
function AnimatedSprite:getCurrentFrameLocalIndex()
```

**Review by Whitebrim — CHANGES_REQUESTED** [review]
Some functions are redundant, you can directly call this properties.

**Comment by dganzella** [comment]
you are right, im Sorry Im new to lua I did not realize this

**Comment by Whitebrim** [comment]
@dganzella You had two usefull functions, please accept all my suggestions and I'll merge it.

**Review by Whitebrim — APPROVED** [review]

**Comment by dganzella** [comment]
ok @Whitebrim I think I accepted them all (I was in the github app and the app does not let me add your changes, so I was a bit confused -- only the website lets me accept changes)

**Pull request merged**